content/blog/how-to-use-nodejs-loader.en.mdx

```diff
@@ -0,0 +1,289 @@
+- **Parameters**:
+  - `specifier`: The string or URL passed to `import` or `require`.
+  - `context`: An object containing information about the context in which the module is being resolved, such as the parent URL and export conditions.
+  - `nextResolve`: A function that calls the next `resolve` hook in the chain, or the default Node.js resolver if this is the last hook.
+- **Returns**: An object containing the resolved URL and optionally the format of the module (e.g., `'module'`, `'commonjs'`).
+
+The `resolve` hook allows you to customize how Node.js finds modules. For example, you could use it to map certain specifiers to different file paths or even to URLs on the web.
+
+#### The `load` Hook
+
+The `load` hook is responsible for loading the content of a module given its resolved URL. Once the `resolve` hook has determined where a module is located, the `load` hook takes over to actually load the module's code.
+
+Here's a breakdown of what the `load` hook does:
+
+- **Function Signature**: `async function load(url, context, nextLoad)`
+- **Parameters**:
+  - `url`: The URL returned by the `resolve` hook.
+  - `context`: An object containing information about the context in which the module is being loaded, such as the format and import attributes.
+  - `nextLoad`: A function that calls the next `load` hook in the chain, or the default Node.js loader if this is the last hook.
+- **Returns**: An object containing the format of the module and the source code to be evaluated.
+
+The `load` hook allows you to customize how Node.js loads modules. For example, you could use it to transpile code from one language to another (e.g., CoffeeScript to JavaScript) or to load modules from non-standard sources.
+
+### Example of a Loader
+
+Let's look at a simple example of a loader that handles `.coffee` files by transpiling them to JavaScript:
+
+```mjs
+// coffee-loader.mjs
+import { readFile } from 'node:fs/promises';
+import { dirname, extname, resolve as resolvePath } from 'node:path';
+import { cwd } from 'node:process';
+import { fileURLToPath, pathToFileURL } from 'node:url';
+import coffeescript from 'coffeescript';
+
+const extensionsRegex = /\.(coffee|litcoffee|coffee\.md)$/;
+
+export async function load(url, context, nextLoad) {
+  if (extensionsRegex.test(url)) {
+    const format = await getPackageType(url);
+    const { source: rawSource } = await nextLoad(url, { ...context, format });
+    const transformedSource = coffeescript.compile(rawSource.toString(), url);
+
+    return {
+      format,
+      shortCircuit: true,
+      source: transformedSource,
+    };
+  }
+
+  return nextLoad(url);
+}
+
+async function getPackageType(url) {
+  const isFilePath = !!extname(url);
+  const dir = isFilePath ? dirname(fileURLToPath(url)) : url;
+  const packagePath = resolvePath(dir, 'package.json');
+  const type = await readFile(packagePath, { encoding: 'utf8' })
```

Comment:
This is a built-in utility now: [`module.findPackageJSON()`](https://nodejs.org/api/module.html#modulefindpackagejsonspecifier-base) (released in `23.2`)

Reply:
same question as `nodejs-loader` this docs is made for LTS so let's leave things as they are

Reply:
It will be in LTS in less than than 2 weeks 🙂

Reply:
okay I'll update